cardinal_service_profile.inc

```diff
@@ -47,3 +49,97 @@ function cardinal_service_profile_node_presave(NodeInterface $node) {
   }
 
 }
+
+/**
+ * Implements hook_preprocess_hook().
+ */
+function cardinal_service_profile_preprocess_flag(&$variables) {
+  if (\Drupal::currentUser()->isAuthenticated()) {
+    return;
+  }
+
+  unset($variables['attributes']['class']);
+  $current_url = \Drupal::request()->getRequestUri();
+  $separator = strpos($current_url, '?') !== FALSE ? '&' : '?';
```

Comment:
I usually use the parse_url() function to check for/pull query params but [it looks strpos actually runs faster](https://stackoverflow.com/a/42805590)